docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,254 @@
+
+This handling of hibernated clusters is a scenario where the [etcd operator-tasks](./05-etcd-operator-tasks.md) framework can be effectively utilized. Therefore, the authors will leverage the operator tasks framework in the design of this solution.
+
+**Implementation Details:**
+
+- **Introduce the `renew-snapshot` Command to `etcdbrctl`:**
+
+  - `etcd-backup-restore` will be enhanced to support a new command `renew-snapshot`, which performs the following steps:
+    - Downloads the latest full snapshot from the object store.
+    - Renames the snapshot by updating the Unix epoch in the filename to reflect the time of the download completion.
+    - Uploads the renamed snapshot back to the object store.
+    - Updates the full snapshot lease after the successful upload.
+
+  The immutability period of an object begins from the moment of upload, so re-uploading the snapshot renews its immutability period. Renaming the snapshot is necessary because uploading with the same name would be considered an attempt to modify an existing snapshot, which is disallowed under immutability policies.
+
+- **Introduce the `garbage-collect` Command to `etcdbrctl`:**
+
+  - `etcd-backup-restore` will be enhanced with a new command `garbage-collect`, which:
+    - Performs garbage collection of snapshots in the object store according to `etcd-backup-restore`'s [garbage collection policy](https://github.com/gardener/etcd-backup-restore/blob/master/docs/usage/garbage_collection.md#gc-policies).
+
+  This functionality is needed to remove old snapshots whose immutability has expired, preventing storage from growing indefinitely.
+
+- **Update the `Etcd` CRD:**
+
+  - Add a new field `immutability.retentionType` under `etcd.spec.backup.store` to specify the type of immutability config.
+
+- **etcd Controller Logic:**
+
+  - When hibernation is requested:
+    - The controller removes the etcd client ports `2380` and `2379` from the `etcd-client` service, leaving only the `etcd-backup-restore` port `8080`, effectively stopping etcd client traffic.
+    - The controller creates an `EtcdOperatorTask` to trigger an on-demand full snapshot.
+    - The controller scales down the etcd cluster by setting `StatefulSet.spec.replicas` to zero.
+    - The controller periodically creates the `ExtendEtcdSnapshotImmutabilityTask` if `etcd.spec.backup.store.immutability.retentionType` is set to `"Bucket"` and based on `etcd.spec.backup.fullSnapshotSchedule`.
+
+- **`ExtendEtcdSnapshotImmutabilityTask` Specification:**
+
+  - Runs `etcdbrctl renew-snapshot` to extend the immutability of the latest snapshot.
+  - Runs `etcdbrctl garbage-collect --garbage-collection-policy <policy>` to remove old snapshots.
+
+- **EtcdOperatorTask Controller Logic:**
+
+  - The operator-tasks controller reacts to the creation of the custom resource and deploys a job named `<etcd-name>-extend-immutability`, which is the `ExtendEtcdSnapshotImmutabilityTask` job.
```

Comment:
there is no change in the operator task controller logic. Just point to the operator task proposal instead.